### ISSUE TITLE
False error reported when using RangeFrom

### ISSUE THREAD

**Issue by andrew-w-ross** [rust-analyzer/rust-analyzer#1542]
Rust analyzer reports expected a block when using a range from expression.

```rust
fn main() {
    for _n in 0..{
        break;
    }// Reports "Syntax Error: expected a block"
    println!("This compiles");
}
```

https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=40a7a97ef6fa291c7b5be312e084b9b7

**Comment by DJMcNab** [comment]
This should be handled using Restrictions.disallow_structs
